README.md

```diff
@@ -12,24 +198,177 @@ command in the console. At the bottom, there will be information similar to the
+
+The roles folder contains the following sub-folders:
+
+[Docker](#docker)
+
+[Installation](#installation)
+
+[Products](#products)
+
+
+## Docker ##
+
+The docker folder contains some general roles that are used in after docker install. It has the following folders:
+
+aws: this is used for installing our shadow-upload.sh script and AWS customer key which uploads ROS logs to AWS. It has a dependency of installation/aws-cli
+
+docker-image: this is used for pulling the docker image (and if nvidia_docker it not 0, it will append -nvidia to the docker image before it is pulled)
+
+setup-ui: this is used to install various UI libraries, terminator, vim, git, subversion, bash-completion, and to create the /usr/local/bin/entrypoint.sh file
+
+## Installation ##
+
+Any programs that need to be installed are placed in the installation role. It has the following folders:
+
+aws-cli
+chrony-client
+chrony-server
+docker
+nvidia-docker
+openvpn-client
+openvpn-server
+pycharm
+qtcreator
+steamvr
+vscode
+
+## Products ##
+
+The Products folder (/ansible/roles/products) contains groupings of roles under folders:
+
+common
+hand-e
+hand-h
+teleop
+
+The logic is: everything to do with hand-e is in the hand-e folder, everything to do with teleop is in the teleop folder.
+The common product is special, see below.
+
+## Common ##
+
+The common role contains any common task or roles that is used repeatedly in many different products. 
+
+It contains the following roles:
+
+cyberglove
+demo-icons
+docker-container
+resources
+save-logs-icons
+
+For example, since docker-container contains the Ansible scripts for creating a Docker container, and it is a very common task shared by many products, it makes sense to have it in common so it can be referred to by other products.
+
+Whenever you create a role, if it will likely be used by other products (e.g. it's not very specific to your product), please put it under common roles, or if it's for installing some commonly used program, put it under installation.
+
+## Dependencies ##
+
+There are 2 main way of including dependencies in Ansible roles. The primary way we use is the "include_role" method because it is dynamic (see [here](https://docs.ansible.com/ansible/latest/modules/include_role_module.html) for documentation).
+
+E.g. if we want to include a particular role to install Docker, we do:
 
 ```bash
-bash <(curl -Ls bit.ly/run-aurora) docker_deploy option1=value1 option2=value2 option3=value3
+- name: Include installation/docker role
+  include_role:
+    name: installation/docker
 ```
+The other way of having dependencies in Ansible is by using the meta folder and main.yml inside the meta folder. Any tasks in meta/main.yml are run before the task/main.yml. An example of meta/main.yml:
```

Comment:
if we do it like that are we going to have issues with defaults overwirting in different computers?

Reply:
Yes. I added a comment about it